tests/test_snapconfig.py

```diff
@@ -1,170 +1,177 @@
-        """Unit test assert equal paths"""
-        self.assertEqual(abspath(p1), abspath(p2))
-
-    def test_FileNotFound(self):
-        snap = SnapConfig.parse(join(self._base, "nosuchwell.sc"))
-        wp_f = snap.filename(0)
-        with self.assertRaises(IOError):
-            WellPath.parse(wp_f)
-
-    def test_ParseMissingGridOrRestart(self):
-        with self.assertRaises(ValueError):
-            SnapConfig.parse(join(self._base, "test-missing-grid.sc"))
-        with self.assertRaises(ValueError):
-            SnapConfig.parse(join(self._base, "test-missing-restart.sc"))
-
-    def test_ParseConf(self):
-        snap = SnapConfig.parse(join(self._base, "littered.sc"))
-        self.assertIsNone(snap.getInit())
-
-        snapinit = SnapConfig.parse(join(self._base, "littered-w-init.sc"))
-        self.assertIsNotNone(snapinit.getInit())
-        self.assertEqualPaths(
-            join(self._ecl_base, "SPE3CASE1.INIT"), snapinit.initFile()
+import yaml
+
+from snapwell import Inf, SnapConfig
+
+testdata_path = join(dirname(__file__), "testdata")
+conf_path = join(testdata_path, "snapwell")
+eclipse_path = join(testdata_path, "eclipse")
+
+
+def same_path(path1, path2):
+    return Path(path1).resolve() == Path(path2).resolve()
+
+
+def test_parse_missing_grid():
+    with open(join(conf_path, "test-missing-grid.yaml")) as config_file:
+        config_dict = yaml.safe_load(config_file)
+    with pytest.raises(TypeError):
+        SnapConfig(**config_dict)
+
+
+def test_parse_missing_restart():
+    with open(join(conf_path, "test-missing-restart.yaml")) as config_file:
+        config_dict = yaml.safe_load(config_file)
+    with pytest.raises(TypeError):
+        SnapConfig(**config_dict)
+
+
+def test_parse_without_init():
+    with open(join(conf_path, "no-init.yaml")) as config_file:
+        config_dict = yaml.safe_load(config_file)
+    conf = SnapConfig(**config_dict)
+    assert conf.init_file is None
+
+
+@pytest.fixture
+def init_config():
+    with open(join(conf_path, "with-init.yaml")) as config_file:
+        config_dict = yaml.safe_load(config_file)
+    return SnapConfig(**config_dict)
+
+
+def test_set_base_path(init_config):
+    init_config.set_base_path(conf_path)
+    assert same_path(join(eclipse_path, "SPE3CASE1.INIT"), init_config.init_file)
+    assert same_path(join(conf_path, "grid.EGRID"), init_config.grid_file)
+    assert same_path(join(conf_path, "restart.UNRST"), init_config.restart_file)
+
+
+def test_wellpath_content(init_config):
+    assert len(init_config.wellpath_files) == 2
+
+    init_config.set_base_path(conf_path)
+    assert same_path(join(conf_path, "well.w"), init_config.wellpath_files[0].well_file)
+    assert same_path(
+        join(conf_path, "well1.w"), init_config.wellpath_files[1].well_file
+    )
+    assert init_config.wellpath_files[0].date == date(2022, 1, 1)
+    assert init_config.wellpath_files[1].date == date(2019, 5, 1)
+
+
+@pytest.fixture
+def full_config():
```

Comment:
Can you dump this instead of reading it here? That way it becomes much easier to review I think.

Reply:
Dumped it, but not sure its any better.